src/pip/_internal/exceptions.py

```diff
@@ -247,3 +247,17 @@ def hash_then_or(hash_name):
 class UnsupportedPythonVersion(InstallationError):
     """Unsupported python version according to Requires-Python package
     metadata."""
+
+
+class UnknownUnstableFeatures(PipError):
+    """Unstable feature names passed are not known.
+    """
+
+    def __init__(self, unknown_names):
+        super(UnknownUnstableFeatures, self).__init__(unknown_names)
```

Comment:
I don't think you need to pass the constructor arguments down to the base class since you're already handling them here. E.g. here's an example in the Python docs of creating a user-defined exception to show that it's not needed: https://docs.python.org/3/tutorial/errors.html#tut-userexceptions

I would also add a test of the `__str__()` method.

Reply:
> I don't think you need to pass the constructor arguments down to the base class since you're already handling them here.

I'd still want this to be initialized like an exception (like having e.args etc).

> I would also add a test of the __str__() method.

Indeed.